src/algebra/algebra/basic.lean

```diff
@@ -1404,20 +1404,28 @@ end linear_map
 end is_scalar_tower
 
 section restrict_scalars
-/- In this section, we describe restriction of scalars: if `S` is an algebra over `R`, then
-`S`-modules are also `R`-modules. -/
 
 section type_synonym
 variables (R A M : Type*)
 
-/--
-Warning: use this type synonym judiciously!
-The preferred way of working with an `A`-module `M` as `R`-module (where `A` is an `R`-algebra),
-is by `[module R M] [module A M] [is_scalar_tower R A M]`.
+/-- A type synonym representing the functor from the category of modules over `A`, regarded as a
+(semi)ring, to the category of modules over `A`, regarded as an `R`-algebra.
 
-When `M` is a module over a ring `A`, and `A` is an algebra over `R`, then `M` inherits a
-module structure over `R`, provided as a type synonym `module.restrict_scalars R A M := M`.
--/
+Warning: use this type synonym judiciously! Often what one wants instead of `restrict_scalars` is
+actually the following setup:
+`[comm_semiring R] [semiring A] [algebra R A] [module R M] [module A M] [is_scalar_tower R A M]` (*)
+which is just the way of saying "let `M` be a module over the `R`-algebra `A`".
```

Comment:
I think a more spelt-out example might be useful here.
````suggestion
Warning: use this type synonym judiciously!

Consider an example where we want to construct an `R`-linear map between `M` and `A`, given
```lean
variables (R A M N : Type*) [comm_semiring R] [semiring A] [algebra R A] [module A M]
```
Often what one wants instead of:
```lean
-- a `R`-module structure on `M` is provided by `restrict_scalars` which is compatible
example : restrict_scalars R A M →ₗ[R] A := sorry
```
is actually the following setup:
```lean
-- a `R`-module structure on `M` and proof of its compatibility is provided by the user
example [module R M] [is_scalar_tower R A M] : M →ₗ[R] A := sorry
```
as in practice concrete `M`s will provide both these typeclasses anyway.
````

Reply:
Yep I think explicit example is very helpful. Updated!